src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,239 @@
+ * trajectory segment
+ *
+ */
+void app_trajectory_planner_interpolate_constant_time_trajectory_segmentation(
+    Trajectory_t* constant_period_trajectory, Trajectory_t* variable_time_trajectory,
+    const float interpolation_period);
+
+
+/**
+ * Returns a the total length of any arc specified by an arc parameterizaton
+ *
+ * @param arc_length_param [in] Arc length parameterization
+ *
+ * @return [out] The arc length in meters
+ */
+static float app_trajectory_planner_get_total_arcLength(
+    ArcLengthParametrization_t arc_length_param);
+
+/**
+ * Returns the maximum velocity profile for a given curve based on curvature and maximum
+ * allowable acceleration
+ *
+ * @pre max_allowable_speed_profile is pre-allocated up the the limit specified by
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param max_allowable_speed_profile [out] The pre-allocated array that will contain all
+ * of the maximum allowable velocity values for each point on th curve
+ * @param traj_elements [in] Trajectory segments including X/Y position
+ *
+ * @param path [in] The polynomial that defines the path of the trajectory
+ *
+ * @param num_elements [in] The number of elements in the trajectory
+ *
+ * @param arc_length_param [in] The arc length parameterization of the path polynomial
+ *
+ * @param arc_segment_length [in] The length of each segment in the trajectory
+ *
+ * @param max_allowable_acceleration [in] The max allowable acceleration at any discrete
+ * point on the trajectory
+ *
+ */
+static void app_trajectory_planner_get_max_allowable_speed_profile(
+    float* max_allowable_speed_profile, TrajectoryElement_t* traj_elements,
+    Polynomial2dOrder3_t path, unsigned int num_elements,
+    ArcLengthParametrization_t arc_length_param, float arc_segment_length,
+    const float max_allowable_acceleration);
+
+/**
+ * Builds the forwards continuous velocity profile for the given parameters.
+ * NOTE: The velocity profile returned by this function is NOT reverse continuous and does
+ * not guarantee a feasible path to follow.
+ *
+ * NOTE: The velocity_profile wil contain the same number of real elements as the input
+ * max_allowable_speed[] array
+ *
+ * @pre The velocity_profile array is pre-allocated to contain up to
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS elements
+ *
+ * @param num_segments [in] The number of segments(elements) in the
+ * max_allowable_speed_profile input array
+ * @param velocity_profile [out] The array that the forwards continuous array will be
+ * copied into
+ * @param max_allowable_speed_profile [in] The pre-allocated array that will contain all
+ * of the maximum allowable velocity values for each point on th curve
+ * @param arc_segment_length [in] The lenmgth of each arc length segment in meters
+ * @param max_allowable_acceleration [in] The max allowable acceleration at any point in
+ * m/s^2
+ * @param max_allowable_speed [IN] The max allowable speed at any point in m/s
```

Comment:
super nit to be consistent with the rest of the inputs
```suggestion
 * @param max_allowable_speed [in] The max allowable speed at any point in m/s
```

Reply:
Lowcase'ed